platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -0,0 +1,305 @@
+        }
+
+        KeyStore trustStore = getKeyStore(System.getProperty("javax.net.ssl.trustStore"),
+                System.getProperty("javax.net.ssl.trustStorePassword"));
+        KeyStore keyStore = getKeyStore(System.getProperty("javax.net.ssl.keyStore"),
+                System.getProperty("javax.net.ssl.keyStorePassword"));
+
+        SSLContext sslContext = null;
+
+        try {
+            sslContext = SSLContexts.custom()
+                    .loadKeyMaterial(keyStore, System.getProperty("javax.net.ssl.keyStorePassword")
+                            .toCharArray())
+                    .loadTrustMaterial(trustStore)
+                    .useTLS()
+                    .build();
+        } catch (UnrecoverableKeyException | NoSuchAlgorithmException | KeyStoreException |
+                KeyManagementException e) {
+            LOGGER.warn("Unable to create secure HttpClient. The Solr Client should not be used in this state.", e);
+            return null;
+        }
+
+        sslContext.getDefaultSSLParameters()
+                .setNeedClientAuth(true);
+        sslContext.getDefaultSSLParameters()
+                .setWantClientAuth(true);
+
+        return sslContext;
+    }
+
+    private static KeyStore getKeyStore(String location, String password) {
+        LOGGER.debug("Loading keystore from {}", location);
+        KeyStore keyStore = null;
+
+        try (FileInputStream storeStream = new FileInputStream(location)) {
+            keyStore = KeyStore.getInstance(System.getProperty("javax.net.ssl.keyStoreType"));
+            keyStore.load(storeStream, password.toCharArray());
+        } catch (CertificateException | IOException
+                | NoSuchAlgorithmException | KeyStoreException e) {
+            LOGGER.warn("Unable to load keystore at {}", location, e);
+        }
+
+        return keyStore;
+    }
+
+    private static void createSolrCore(String url, String coreName, String configFileName,
+            HttpClient httpClient) throws IOException, SolrServerException {
+        HttpSolrClient client;
+        if (httpClient != null) {
+            client = new HttpSolrClient(url, httpClient);
+        } else {
+            client = new HttpSolrClient(url);
+        }
+
+        HttpResponse ping = client.getHttpClient()
+                .execute(new HttpHead(url));
+        if (ping != null && ping.getStatusLine()
+                .getStatusCode() == 200) {
+            ConfigurationFileProxy configProxy =
+                    new ConfigurationFileProxy(ConfigurationStore.getInstance());
+            configProxy.writeSolrConfiguration(coreName);
+            if (!solrCoreExists(client, coreName)) {
+                LOGGER.debug("Creating Solr core {}", coreName);
+
+                String configFile = StringUtils.defaultIfBlank(configFileName,
+                        DEFAULT_SOLRCONFIG_XML);
+
+                String solrDir;
+                if (System.getProperty("solr.data.dir") != null) {
```

Comment:
It looks like the overall DDF codebase is a mix of using string literals and constants for system property keys.  I will leave them hardcoded here since these values should not be changed.  If they were, it would impact system upgrades.